File: frontend/src/types/cluster.ts
export interface NodeCapacityDetail {
  node_name: string;
  cpu_allocatable_cores: number;
  cpu_requested_cores: number;
  cpu_used_cores: number;
  memory_allocatable_bytes: number;
  memory_requested_bytes: number;
  memory_used_bytes: number;
  ready: boolean;
  schedulable: boolean;
}

export interface ClusterCapacity {
  cpu_allocatable_cores: number;
  cpu_requested_cores: number;
  cpu_used_cores: number;
  memory_allocatable_bytes: number;
  memory_requested_bytes: number;
  memory_used_bytes: number;
  nodes: NodeCapacityDetail[];
}

File: backend/app/schemas/cluster.py
from pydantic import BaseModel


class NodeCapacityDetail(BaseModel):
    node_name: str
    cpu_allocatable_cores: float
    cpu_requested_cores: float
    cpu_used_cores: float
    memory_allocatable_bytes: int
    memory_requested_bytes: int
    memory_used_bytes: int
    ready: bool
    schedulable: bool


class ClusterCapacity(BaseModel):
    cpu_allocatable_cores: float
    cpu_requested_cores: float
    cpu_used_cores: float
    memory_allocatable_bytes: int
    memory_requested_bytes: int
    memory_used_bytes: int
    nodes: list[NodeCapacityDetail]
